javascript/require-and-import-mini-lesson.md

```diff
@@ -0,0 +1,351 @@
+- [JavaScript 5 - Switch Statements Lesson](/javascript/javascript-5-switch.md)
+- [JavaScript 6 - Object Literals Lesson](/javascript/javascript-6-object-literals.md)
+- [JavaScript 7 - Object-Oriented Programming](/javascript/javascript-7-oop.md)
+
+### Motivation
+
+Why it is worth learning this topic
+
+- Knowing when to use Require and when using Import is very helpful when building a full stack application.
+- This mini lesson will also help you understand the differences of using each while building [web APIs](https://github.com/Techtonica/curriculum#week-6---web-apis), when doing the [Eventonica with React](https://github.com/Techtonica/curriculum#week-8---eventonica-with-react) section and also the [PERN Mini-Project](https://github.com/Techtonica/curriculum#week-9---pern-mini-project-postgresexpressreactnode)
+- In a technical interview, you will be confident to answer the question: `When should you use Import instead of Require?`
+
+### Objectives
+
+**Participants will be able to:**
+
+- Differentiate between `require` and `import`
+- Be initiated to modular programming
+
+### Specific Things To Learn
+
+- [What are JavaScript modules?](#what-are-javascript-modules)
+- [What is require?](#what-is-require)
+- [What is import?](#what-is-import)
+- [Difference between import and require](#difference-between-import-and-require)
+- [Can we use import in node?](#can-we-use-import-in-node)
+- [Which one to use? And When?](#which-one-to-use?-and-when?)
+- [Chart of the Differences](#chart-of-the-differences)
+
+
+### Lesson
+
+With JavaScript, when you want to include code that another developer has produced and made available to the DEV community, you now two main ways to bring in that code into your own script: `require` and `import`. When starting out, it can sometimes become confusing when to use one or another. So, in order to clarify this a little bit, let's dive into the core of what they are, how they differ, and when you should use one or the other.
+
+#### What are JavaScript modules?
+
+The first thing we need to do is to understand what a JavaScript module is. Basically, it's a piece of code which is *exporting* objects that you can import in your own code. So, yes, `require` imports, and `import`, well, you've guessed it alright. The only thing is that they do it differently.
+
+If you want more information on modules, I invite you to have a look at the official MDN documentation in the [supplemental materials section](#supplemental-materials).
+
+#### What is require?
+
+In order to deal with JavaScript outside of the browser, Node.js follows the CommonJS module specification in which modules are loaded synchronously, i.e. one after the other. This is why you cannot use `require` as such in the browser, unless it is transpiled. A code is said to be transpiled when it is converted from a certain source code to another source code but producing the equivalent functionalities.
```

Comment:
This might be too much detail. Browsers simply don't have a `window.require` and that might be enough to know for now. I don't think sync vs async is a necessary distinction. I also wouldn't bring up transpilation unless needed, particularly because browsers can now run es6 directly and so the transpilation down to es5 is less important than just bundling it all together.

Reply:
done in same commit